pkg/server/sandbox_run.go

```diff
@@ -303,8 +308,61 @@ func (c *criContainerdService) setupSandboxFiles(rootDir string, config *runtime
 	if err := c.os.CopyFile(etcHosts, sandboxEtcHosts, 0666); err != nil {
 		return fmt.Errorf("failed to generate sandbox hosts file %q: %v", sandboxEtcHosts, err)
 	}
-	// TODO(random-liu): [P0] Set DNS options. Maintain a resolv.conf for the sandbox.
+	// Set DNS options. Maintain a resolv.conf for the sandbox.
+	resolvContent := []byte{}
+	var err error
+	if dnsConfig := config.GetDnsConfig(); dnsConfig != nil {
+		resolvContent, err = parseDNSOptions(dnsConfig.Servers, dnsConfig.Searches, dnsConfig.Options)
+		if err != nil {
+			return fmt.Errorf("failed to parse sandbox dnsConfig: %v", err)
+		}
+	}
+	resolvPath := getResolvPath(rootDir)
+	hostNetwork := config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetHostNetwork()
+	if len(resolvContent) == 0 && hostNetwork {
+		// copy host's resolv.conf to resolvPath
+		err = c.os.CopyFile(resolvConfPath, resolvPath, 0755)
+	} else {
+		err = c.os.WriteFile(resolvPath, resolvContent, 0755)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to copy/write file to %q: %v", resolvPath, err)
+	}
+
 	// TODO(random-liu): [P0] Deal with /dev/shm. Use host for HostIpc, and create and mount for
 	// non-HostIpc. What about mqueue?
 	return nil
 }
+
+// parseDNSOptions parse DNS options into resolv.conf format content,
+// if none option is specified, will return empty with no error.
+func parseDNSOptions(servers, searches, options []string) ([]byte, error) {
```

Comment:
nit: Can we use `string` in this function, and use `resolvContent += fmt.Sprintf("search %s\n", strings.Join(searches, " "))`, which is much more cleaner.